api/src/main/java/org/apache/iceberg/ExpireSnapshots.java

```diff
@@ -118,4 +118,9 @@ public interface ExpireSnapshots extends PendingUpdate<List<Snapshot>> {
    * @return this for method chaining
    */
   ExpireSnapshots cleanExpiredFiles(boolean clean);
+
+  /** Returns number of expired snapshots */
+  default long expiredSnapshotsCount() {
+    return 0L;
```

Comment:
Hm, I think it would be weird to have 0 as the number of expired snapshots yet have deleted files at the same time. We can consider throwing an unsupported exception or returning null, however, I am not sure will be any better. My initial thought would be to throw an exception but I need to think more about it.

What do you think, @manuzhang @nastra?

Reply:
Yes, it doesn't make sense when `cleanExpiredFiles(true)` is called or an implementation deletes files by default. I think throw exception is better.

Reply:
I'd probably lean towards an exception.

Reply:
Actually, I thought it was an action class. I am still worried about exposing this stateful method in our public API. I wonder if offering some sort of commit hooks that would have access to the state before and after would be a better idea. 

@rdblue @nastra @manuzhang @amogh-jahagirdar, any thoughts? I guess we can compute this one in particular externally by comparing the number of snapshots before and after the commit call, but it seems like a common problem.